packages/babel-traverse/src/scope/index.js

```diff
@@ -740,6 +743,10 @@ export default class Scope {
     path.traverse(collectorVisitor, state);
     this.crawling = false;
 
+    if (state.unnamedDefault) {
+      state.unnamedDefault.id = this.generateUidIdentifier("default");
```

Comment:
I don't think this is the responsibility of `babel-traverse`. The places that work with declarations should be the ones deciding what do it. If we want, we could have a helper either inside https://github.com/babel/babel/blob/master/packages/babel-helper-function-name/src/index.js, or at least similar to it. Things would explicitly call out to to name the class.